### ISSUE TITLE
FileLocked exception when using memcache for locking

### ISSUE THREAD

**Issue by stephan2012** [owncloud/core#24084]
Dear ownCloud maintainers,

when setting memcache.locking to Redis the creation of new files or directories fails:

```
Exception: {"Message":"HTTP\/1.1 423 \"blub\" is locked","Exception":"OCA\\DAV\\Connector\\Sabre\\Exception\\FileLocked","Code":0,"Trace":"#0 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(1150): OCA\\DAV\\Connector\\Sabre\\Directory->createDirectory('blub')\n#1 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/CorePlugin.php(588): Sabre\\DAV\\Server->createCollection('blub', Object(Sabre\\DAV\\MkCol))\n#2 [internal function]: Sabre\\DAV\\CorePlugin->httpMkcol(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#3 \/var\/www\/owncloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#4 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(459): Sabre\\Event\\EventEmitter->emit('method:MKCOL', Array)\n#5 \/var\/www\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(248): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#6 \/var\/www\/owncloud\/apps\/dav\/appinfo\/v1\/webdav.php(55): Sabre\\DAV\\Server->exec()\n#7 \/var\/www\/owncloud\/remote.php(138): require_once('\/var\/www\/ownclo...')\n#8 {main}","File":"\/var\/www\/owncloud\/apps\/dav\/lib\/connector\/sabre\/directory.php","Line":173,"User":"XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"}
```

This is my config:

```
  'redis' => 
  array (
    'host' => 'localhost',
    'port' => 6379,
  ),
  'memcache.local' => '\OC\Memcache\APCu',
  'memcache.locking' => '\OC\Memcache\Redis',

```

When commenting out the 'memcache.locking' line, everything works as expected.The Redis process is up&running an listening on port 6379.

### Server configuration
**Operating system**: Debian GNU/Linux 8.4

**Web server:** Apache 2.4.18-2 (Debian provided)

**Database:** MariaDB 10.1.13

**PHP version:** 7.0.5-2 (from Debian testing)

**ownCloud version:** 9.0.1

**Updated from an older ownCloud or fresh install:** fresh install

**Where did you install ownCloud from:** Official packages

**Signing status (ownCloud 9.0 and above):** valid (no errors found)

```
No errors have been found.
```

Thanks for your support.

Best regards

Stephan

**Comment by MorrisJobke** [comment]
> when setting memcache.locking to Redis the creation of new files or directories fails:

Could you create another file? (with a different name)

**Comment by stephan2012** [comment]
No, I cannot create any file or directory. I've tried via a WebDAV mount, too. But no success. Additionally I've just noticed that I cannot change existing files. Looks like any write access is blocked. Workaround for me at the moment is to just disable memcache.locking.

**Comment by MorrisJobke** [comment]
Mmmh. The stray locks has been fixed in 8.2.2 and 9.0.0 - https://github.com/owncloud/core/issues/20380

Could you clear the redis cache and try again? Maybe it's some leftover from before 🙈

**Comment by stephan2012** [comment]
Thanks for the pointer! :-)

Indeed, wiping the Redis cache solved my issue. I can now create create and write to files and directories. Issue solved, thanks again for your help.
